src/core/parse_gal.cpp

```diff
@@ -0,0 +1,10 @@
+#include "parse_gal.hpp"
+
+namespace core {
+	option<std::vector<work_unit>> parse_gal(const std::string & filename) {
+		// TODO:
+		// divide gal into individual units or groups of work
+		// Currently return empty object, indicating failure
+		return {};
+	}
+}
```

Comment:
This function would parse a gal file into groupings of regions of the tiff file to work on. Dividing the data this way will make the problem easier to parallelize.